addon/session-stores/local-storage.js

```diff
@@ -39,24 +39,30 @@ export default BaseStore.extend({
 
     @method persist
     @param {Object} data The data to persist
+    @return {Ember.RSVP.Promise} A promise that resolves when the data has been persisted and rejects otherwise.
     @public
   */
   persist(data) {
-    data = JSON.stringify(data || {});
-    localStorage.setItem(this.key, data);
-    this._lastData = this.restore();
+    return new RSVP.Promise((resolve) => {
+      data = JSON.stringify(data || {});
+      localStorage.setItem(this.key, data);
+      this.restore().then((restoredContent) => {
+        this._lastData = restoredContent;
+        resolve();
+      }, resolve);
+    });
```

Comment:
This is the same as above for the cookie store basically - if we don't use `restore` here to set `_lastData` we don't have to synchronize restoration and actually writing the data.